packages/transaction-controller/src/TransactionController.ts

```diff
@@ -642,7 +632,25 @@ export class TransactionController extends BaseControllerV1<
     });
 
     this.onBootCleanup();
-    this.#initTrackingMap();
+
+    if (this.enableMultichain) {
+      this.messagingSystem.subscribe(
+        'NetworkController:stateChange',
```

Comment:
Could we add one more small test just verifying that none of the events associated with `refreshTrackingMap` or `initTrackingMap` are called when a stateChange event occurs and the feature flag is off?

Reply:
we should add the `this.enableMultichain` guard inside of the subscribe rather than guarding the entire subscribe if we care about being able to toggle this on and off without having to restart extension.

Reply:
unclear if we will want this to be toggleable at runtime but might as well!